crates/dojo-erc/src/erc721/erc721.cairo

```diff
@@ -118,29 +143,75 @@ mod ERC721 {
 
     #[external(v0)]
     fn set_approval_for_all(ref self: ContractState, operator: ContractAddress, approved: bool) {
+        let token = get_contract_address();
+        let caller = get_caller_address();
+
         let mut calldata = ArrayTrait::new();
-        let owner = get_caller_address();
-        assert(owner != operator, 'ERC721: approval to owner');
-        calldata.append(owner.into());
+        calldata.append(token.into());
+        calldata.append(caller.into());
         calldata.append(operator.into());
         calldata.append(approved.into());
         self.world.read().execute('erc721_set_approval_for_all'.into(), calldata);
-        self.emit(ApprovalForAll { owner, operator, approved });
+        self.emit(ApprovalForAll { owner: caller, operator, approved });
+    }
+
+    #[external(v0)]
+    fn transfer(ref self: ContractState, to: ContractAddress, token_id: u256) {
+        transfer_from(ref self, get_caller_address(), to, token_id);
+    }
+
+    #[external(v0)]
+    fn safe_transfer_from(
```

Comment:
safe transfer isn't needed with account abstraction. let's remove this method. see: https://github.com/starknet-io/SNIPs/discussions/36#discussioncomment-6640087

Reply:
what happen if you want to send an nft to a contract & run some code on transfer ?

[IERC721Receiver.onERC721Received](https://docs.openzeppelin.com/contracts/4.x/api/token/erc721#IERC721Receiver-onERC721Received-address-address-uint256-bytes-)